src/main/java/net/neoforged/neoforge/items/ItemHandlerHelper.java

```diff
@@ -185,4 +191,84 @@ public static int calcRedstoneFromInventory(@Nullable IItemHandler inv) {
             return Mth.floor(proportion * 14.0F) + (itemsFound > 0 ? 1 : 0);
         }
     }
+
+    /**
+     * Try to move items from one inventory to another.
+     *
+     * @param from          source inventory
+     * @param to            destination inventory
+     * @param filter        a filter to select which items to move; if the filter returns {@code false} for a stack, it won't be moved
+     * @param maxAmount     maximum total amount of items to move
+     * @param stackInTarget {@code true} to try to stack with existing items in the target inventory,
+     *                      {@code false} to insert in the first slot that fits
+     * @return the total amount of moved items
+     */
+    public static int moveItems(
+            @Nullable IItemHandler from,
+            @Nullable IItemHandler to,
+            Predicate<ItemStack> filter,
+            int maxAmount,
+            boolean stackInTarget) {
+        Objects.requireNonNull(filter);
```

Comment:
Dunno how consistently we do this, but IMHO in APIs we should provide the name of the argument that was null...

```suggestion
        Objects.requireNonNull(filter, "filter");
```